diff --git desktop/Makefile.am desktop/Makefile.am
index bc13f46..4fcce4b 100644
--- desktop/Makefile.am
+++ desktop/Makefile.am
@@ -13,10 +13,12 @@ svg_DATA = azpainter.svg application-x-azpainter-apd.svg
 mimetypedir = $(datadir)/mime/packages
 mimetype_DATA = azpainter.xml
 
+appname = $(shell sed -n '/^Name=/s///p' azpainter.desktop).app
+
 ### rule
 
-install-data-hook: update-cache
-uninstall-hook: update-cache
+install-data-hook: update-cache install-for-mac
+uninstall-hook: update-cache uninstall-for-mac
 
 update-cache:
 	@-if test -z "$(DESTDIR)"; then \
@@ -25,3 +27,13 @@ update-cache:
 		update-mime-database $(datadir)/mime; \
 		update-desktop-database $(datadir)/applications; \
 	fi
+
+install-for-mac:
+	echo "do shell script \"\
+		LANG=$(shell defaults read -g AppleLocale | sed 's/@.*$$//g').UTF-8\
+		$(DESTDIR)$(bindir)/$(PACKAGE) >/dev/null 2>&1 &\"" | osacompile -o $(appname)
+	makeicns -in azpainter.png -out $(appname)/Contents/Resources/applet.icns
+	cp -rf $(appname) /Applications/
+
+uninstall-for-mac:
+	rm -rf /Applications/$(appname)
diff --git desktop/Makefile.in desktop/Makefile.in
index 09fbdeb..8e3d7ec 100644
--- desktop/Makefile.in
+++ desktop/Makefile.in
@@ -249,6 +249,7 @@ svgdir = $(datadir)/icons/hicolor/scalable/apps
 svg_DATA = azpainter.svg application-x-azpainter-apd.svg
 mimetypedir = $(datadir)/mime/packages
 mimetype_DATA = azpainter.xml
+appname = $(shell sed -n '/^Name=/s///p' azpainter.desktop).app
 all: all-am
 
 .SUFFIXES:
@@ -530,8 +531,8 @@ uninstall-am: uninstall-desktopDATA uninstall-iconDATA \
 
 ### rule
 
-install-data-hook: update-cache
-uninstall-hook: update-cache
+install-data-hook: update-cache install-for-mac
+uninstall-hook: update-cache uninstall-for-mac
 
 update-cache:
 	@-if test -z "$(DESTDIR)"; then \
@@ -541,6 +542,16 @@ update-cache:
 		update-desktop-database $(datadir)/applications; \
 	fi
 
+install-for-mac:
+	echo "do shell script \"\
+		LANG=$(shell defaults read -g AppleLocale | sed 's/@.*$$//g').UTF-8\
+		$(DESTDIR)$(bindir)/$(PACKAGE) >/dev/null 2>&1 &\"" | osacompile -o $(appname)
+	makeicns -in azpainter.png -out $(appname)/Contents/Resources/applet.icns
+	cp -rf $(appname) /Applications/
+
+uninstall-for-mac:
+	rm -rf /Applications/$(appname)
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff --git mlib/src/widget/mWindow.c mlib/src/widget/mWindow.c
index 0cacf19..4e62ee0 100644
--- mlib/src/widget/mWindow.c
+++ mlib/src/widget/mWindow.c
@@ -303,7 +303,7 @@ void mWindowMoveCenter(mWindow *p,mWindow *win)
 	if(x < box.x) x = box.x;
 	if(y < box.y) y = box.y;
 
-	mWidgetMove(M_WIDGET(p), x, y);
+	mWidgetMove(M_WIDGET(p), 0, y);
 }
 
 /** 初期表示処理
diff --git src/Makefile.am src/Makefile.am
index a203c43..e5c46cc 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -3,7 +3,7 @@ AM_CFLAGS=-pthread
 
 bin_PROGRAMS=azpainter
 azpainter_LDADD=../mlib/libmlib.a \
-  -lX11 -lXext -lfontconfig -lfreetype -lpng -ljpeg -lz -lrt -lm
+  -lX11 -lXext -lfontconfig -lfreetype -lpng -lturbojpeg -lz -lm
 
 azpainter_SOURCES= \
 AppCursor.c \
diff --git src/Makefile.in src/Makefile.in
index c9c3bd4..c45d421 100644
--- src/Makefile.in
+++ src/Makefile.in
@@ -345,7 +345,7 @@ top_srcdir = @top_srcdir@
 AM_CPPFLAGS = -I../mlib/include -Iinclude -I@freetype_dir@ -DPACKAGE_DATA_DIR=\""$(datadir)/azpainter"\"
 AM_CFLAGS = -pthread
 azpainter_LDADD = ../mlib/libmlib.a \
-  -lX11 -lXext -lfontconfig -lfreetype -lpng -ljpeg -lz -lrt -lm
+  -lX11 -lXext -lfontconfig -lfreetype -lpng -lturbojpeg -lz -lm
 
 azpainter_SOURCES = \
 AppCursor.c \
