diff --git configure configure
index cb9a9f6..565c280 100755
--- configure
+++ configure
@@ -226,8 +226,8 @@ tmp1=`uname`
 if test "$tmp1" = Linux;then cf_os=linux
 elif test "$tmp1" = Darwin;then
 	cf_os=mac
-	CFLAGS=`fc_add_string "$CFLAGS" "-I/usr/X11/include"`
-	LDFLAGS=`fc_add_string "$LDFLAGS" "-L/usr/X11/lib"`
+	CFLAGS=`fc_add_string "$CFLAGS" "-I/opt/X11/include"`
+	LDFLAGS=`fc_add_string "$LDFLAGS" "-L/opt/X11/lib"`
 elif test "$tmp1" = FreeBSD;then
 	cf_os=freebsd
 	cf_make=gmake
@@ -311,7 +311,7 @@ fc_check_lib libjpeg "libjpeg" "" "-ljpeg" "stdio.h jpeglib.h" "libjpeg-dev or l
 fc_check_lib X11 "x11" "" "-lX11" "X11/Xlib.h" "libx11-dev or libX11-devel or libx11"
 fc_check_lib Xext "xext" "" "-lXext" "X11/Xlib.h X11/extensions/XShm.h" "libxext-dev or libXext-devel or libxext"
 fc_check_lib XInput "xi" "" "-lXi" "X11/Xlib.h X11/extensions/XInput2.h" "libxi-dev or libXi-devel or libxi"
-fc_check_lib fontconfig "fontconfig" "-I/usr/include/freetype2 -I/usr/X11/include/freetype2 -I/usr/local/include/freetype2" "-lfontconfig -lfreetype" "fontconfig/fontconfig.h" "libfontconfig1-dev or libfontconfig-devel or fontconfig"
+fc_check_lib fontconfig "fontconfig" "-I/usr/include/freetype2 -I/opt/X11/include/freetype2 -I/usr/local/include/freetype2" "-lfontconfig -lfreetype" "fontconfig/fontconfig.h" "libfontconfig1-dev or libfontconfig-devel or fontconfig"
 fc_check_lib freetype2 "freetype2" "" "" "ft2build.h" "libfreetype6-dev or libfreetype6-devel or freetype2"
 
 #------------------
diff --git desktop/icons/hicolor/scalable/apps/azpainter.svg desktop/icons/hicolor/scalable/apps/azpainter.svg
index 023e08a..d2dc659 100755
--- desktop/icons/hicolor/scalable/apps/azpainter.svg
+++ desktop/icons/hicolor/scalable/apps/azpainter.svg
@@ -167,7 +167,7 @@
        x="20.971798"
        y="560.68152"
        sodipodi:insensitive="true"
-       ry="44.807964"
+       ry="0"
        rx="0" />
     <g
        id="g4320"
